inotify/inotify_others.go

```diff
@@ -0,0 +1,52 @@
+// +build !linux
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package inotify // import "k8s.io/utils/inotify"
+
+import (
+	"fmt"
+	"runtime"
+)
+
+// NewWatcher creates and returns a new inotify instance using inotify_init(2)
+func NewWatcher() (*Watcher, error) {
+	return nil, fmt.Errorf("watch not supported on %s", runtime.GOOS)
```

Comment:
possibly the error can shared across all functions.

```
var notSupportedError = fmt.Errorf("watch not supported on %s", runtime.GOOS)
...

func NewWatcher() (*Watcher, error) {
	return nil, notSupportedError 
}
...
```

Reply:
DONE!